src/analyses/dirty.cpp

```diff
@@ -19,14 +19,49 @@ Date: March 2013
 void dirtyt::build(const goto_functiont &goto_function)
 {
   for(const auto &i : goto_function.body.instructions)
-    i.apply([this](const exprt &e) { find_dirty(e); });
+  {
+    if(i.is_other())
+    {
+      search_other(i);
+    }
+    else
+    {
+      i.apply([this](const exprt &e) { find_dirty(e); });
+    }
+  }
+}
+
+void dirtyt::search_other(const goto_programt::instructiont &instruction)
+{
+  INVARIANT(instruction.is_other(), "instruction type must be OTHER");
```

Comment:
Pedantry but I would `PRECONDITION` this because it is the caller's responsibility rather than a property of the algorithm.